src/tls.c

```diff
@@ -7657,7 +7657,7 @@ static int TLSX_KeyShare_ProcessEcc(WOLFSSL* ssl, KeyShareEntry* keyShareEntry)
     }
     if (keyShareEntry->key) {
         wc_ecc_free((ecc_key*)keyShareEntry->key);
-        XFREE(keyShareEntry->key, ssl->heap, DYNAMIC_TYPE_ECC);
+        XFREE(keyShareEntry->key, ssl->heap, DYNAMIC_TYPE_PRIVATE_KEY);
```

Comment:
Good catch! Perhaps it would be better to change the alloc to `DYNAMIC_TYPE_ECC`? For customers using  `DYNAMIC_TYPE_ECC` they would expect this to be the `sizeof(ecc_key)`, which this is.

7045:
```
/* Allocate an ECC key to hold private key. */
kse->key = (byte*)XMALLOC(sizeof(ecc_key), ssl->heap, DYNAMIC_TYPE_PRIVATE_KEY);
```

Reply:
Changed to DYNAMIC_TYPE_ECC.
Free changed back and allocate changed to ECC.